hera_sim/tests/test_sigchain.py

```diff
@@ -457,13 +460,144 @@ def test_mutual_coupling():
                 bl_len = np.linalg.norm(enu_antpos[ak] - enu_antpos[ai])
                 exp_amp = vis_amps[(ak, aj)] * refl_amp / bl_len
                 actual_amp = np.abs(vis_fft[kj_ind])
-                print(f"{(ak,aj)} coupled into {(ai,aj)}")
-                print(f"Expected amplitude: {exp_amp}")
-                print(f"Simulated amplitude: {actual_amp}")
-                print("\n")
                 assert np.isclose(exp_amp, actual_amp, atol=1e-7, rtol=0.05)
 
 
+@pytest.fixture
+def sample_uvdata():
+    hera_sim.defaults.set("debug")
+    uvdata = empty_uvdata()
+    hera_sim.defaults.deactivate()
+    return uvdata
+
+
+@pytest.fixture
+def sample_coupling(sample_uvdata):
+    coupling = sigchain.MutualCoupling(
+        uvbeam="uniform",
+        ant_1_array=sample_uvdata.ant_1_array,
+        ant_2_array=sample_uvdata.ant_2_array,
+        pol_array=sample_uvdata.polarization_array,
+        array_layout=dict(zip(*sample_uvdata.get_ENU_antpos()[::-1])),
+    )
+    return coupling
+
+
+@pytest.fixture
+def uvbeam(tmp_path):
+    beam_fn = str(tmp_path / "test_beam.fits")
+    beam = AnalyticBeam("uniform")
+
+    # Setup some things needed to mock up the UVBeam object
+    az = np.linspace(0, 2 * np.pi, 100)
+    za = np.linspace(np.pi / 2 - np.pi / 6, np.pi / 2 + np.pi / 6, 15)
+    freqs = np.linspace(99e6, 101e6, 20)[None, :]
+    data_shape = (2, 1, 2, freqs.size, za.size, az.size)
+    basis_shape = (2, 2, za.size, az.size)
+    az_mesh, za_mesh = np.meshgrid(az, za)
+
+    # Populate a UVBeam object with everything needed to write to disk...
+    uvb = UVBeam()
+
+    # Starting with the string-type metadata
+    uvb.antenna_type = "simple"
+    uvb.beam_type = "efield"
+    uvb.data_normalization = "peak"
+    uvb.feed_name = "test"
+    uvb.feed_version = "0.0"
+    uvb.history = ""
+    uvb.model_name = ""
+    uvb.model_version = ""
+    uvb.pixel_coordinate_system = "az_za"
+    uvb.telescope_name = "test"
+
+    # Next onto the array-like objects
+    uvb.axis1_array = az
+    uvb.axis2_array = za
+    uvb.bandpass_array = np.ones_like(freqs)
+    uvb.basis_vector_array = np.zeros(basis_shape)
+    uvb.basis_vector_array[0, 0, :, :] = 1
+    uvb.basis_vector_array[1, 1, :, :] = 1
+    uvb.data_array = (
+        beam.interp(az_mesh.flatten(), za_mesh.flatten(), freqs.flatten())[0]
+        .reshape(data_shape)
+        .astype(complex)
+    )
+    uvb.freq_array = freqs
+    uvb.feed_array = np.array(["x", "y"])
+    uvb.spw_array = np.zeros(1, dtype=int)
+
+    # Finally, all the shape parameters
+    uvb.Naxes1 = az.size
+    uvb.Naxes2 = za.size
+    uvb.Naxes_vec = 2
+    uvb.Ncomponents_vec = 2
+    uvb.Nfeeds = 2
+    uvb.Nfreqs = freqs.size
+    uvb.Nspws = 1
```

Comment:
So, all this could essentially be achieved with a `AnalyticBeam.to_uvbeam()` method?

Reply:
Yes, though you'd need to provide an `az_array`, `za_array`, and `freq_array`, and then all the other stuff (like the `feed_name` and whatnot) could be optional parameters.

Reply:
Yeah, I think this is in the works in pyuvdata, but for now this is good.

Reply:
Oh, nice! That would be super convenient.